tools/scripts/checkCommitMsg.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from 'fs';
+
+let msgPath = process.argv[2]
+let msg = fs.readFileSync(msgPath, 'utf-8')
+
+let msgArr = msg.split(' ')
+let scope = msgArr[0]
+let project = msgArr[1]
+let issue = msgArr[msgArr.length - 1]
+  .replace(/\n/, ``)
+let summary = msg.replace(scope, ` `)
+  .replace(project, ``)
+  .replace(issue, ``)
+  .replace(/ /g, ``)
+let isValid = true
+
+if(!/:.+:/.test(scope)) {
+  console.log(`Scope ${scope} is not valid`)
+  isValid = false
+}
+
+if(!/\[.+]:$/.test(project)) {
```

Comment:
Should limit to existing projects/libraries, currently: root, mailer, evidence, contracts, docs

Reply:
Ok, that's a more strict way :)

Reply:
After some further thought, I would like to define a set of app/library types that are used instead of `project` name for commit messages and general organization. For example, `web`, `api`, `blockchain`, `docs`. The project breakdown would be:
- web
  - Evidence uploader frontend
  - Plugin
- api
  - Evidence uploader backend
- blockchain
  - Smart contracts
  - Blockchain mailer
- docs
  - Developer docs
  - User/merchant docs
- root

The blockchain mailer could fit in either `blockchain`, `api`, or even a separate category for utilities/services. I'm not too concerned about the exact details, but this method of organization will more closely match our other projects.